resources/build/bootstrap-cibuildwheel-manylinux-2014.sh

```diff
@@ -0,0 +1,31 @@
+#!/usr/bin/env bash
+
+set -xeo pipefail
+
+# Install additional build tools.
+pip3 install -r "resources/build/requirements.txt"
+# Use explicit predictable conan root path, where packages are 
+# installed.
+export CONAN_USER_HOME="$HOME/conan"
+
+# Create default conan profile so we can configure it before instlibXcomposite-develall.
+# Use --force so that if it already exists we don't error out.
+conan profile new default --detect --force
+# Use old C++11 ABI as per VFX Reference Platform CY2022. Not strictly
+# necessary as this is the default for conan, but we can't be certain
+# it'll remain the default in future.
+conan profile update settings.compiler.libcxx=libstdc++ default
+
+# This script not runnable from outside a cibuildwheel environment.
+if [ "$CIBUILDWHEEL" != "1" ]
+then
+  echo "Error. Attempting to run cibuildwheel-manylinux bootstrap outside of cibuildwheel"
+  exit 1
+fi
```

Comment:
[question] What breaks if its not invoked by `cibuildwheel`? Should we error at the top instead if it is broken to avoid partially polluting the calling env?

Reply:
This was mostly a scope creep avoidance move to stop us having to figure out that very question, as this was originally a `bootstrap-centos` script and it got more specific.

Aye, this should be right at the beginning, will move.

Reply:
[34e1b43](https://github.com/OpenAssetIO/OpenAssetIO/pull/665/commits/34e1b437c487b640aed292abb2e40acc75d1200a)